testsuite/smoke-test/src/consensus/optqs_fault_tolerance.rs

```diff
@@ -0,0 +1,103 @@
+use crate::{
+    consensus::helpers::generate_traffic_and_assert_committed, smoke_test_environment::SwarmBuilder,
+};
+use aptos_forge::{wait_for_all_nodes_to_catchup, NodeExt, SwarmExt};
+use aptos_rest_client::Client;
+use std::{sync::Arc, time::Duration};
+
+const MAX_WAIT_SECS: u64 = 60;
+
+/// This test asserts whether a node that cannot reach either the batch or the block author
+/// can fetch batches from the block voters.
+#[tokio::test]
+async fn test_remote_batch_read_from_block_voters() {
+    let mut swarm = SwarmBuilder::new_local(4)
+        .with_aptos()
+        .with_init_config(Arc::new(|_, conf, _| {
+            conf.api.failpoints_enabled = true;
+            conf.consensus.quorum_store.enable_opt_quorum_store = true;
+        }))
+        .build()
+        .await;
+    let validator_peer_ids = swarm.validators().map(|v| v.peer_id()).collect::<Vec<_>>();
+
+    swarm
+        .wait_for_all_nodes_to_catchup(Duration::from_secs(MAX_WAIT_SECS))
+        .await
+        .unwrap();
+
+    let validator_client_0 = swarm
+        .validator(validator_peer_ids[0])
+        .unwrap()
+        .rest_client();
+    let validator_client_3 = swarm
+        .validator(validator_peer_ids[3])
+        .unwrap()
+        .rest_client();
+
+    // Make every node fetch the batch
+    validator_client_0
+        .set_failpoint(
+            "consensus::send::broadcast_self_only".to_string(),
+            "return(BatchMsg)".to_string(),
```

Comment:
what does this mean? this is the arg passing to the closure right? or it's only the "BatchMsg" part passing to the closure?

Reply:
"BatchMsg" is passed to the closure. return is the failpoint mode - it evaluates the closure and returns the value returned by the closure.